src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.person.predicates;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+import seedu.address.model.util.IncomeComparisonOperator;
+
+/**
+ * Predicate that compares a {@code Person}'s income against a threshold using a specified comparison operator.
+ */
+public class IncomeComparisonPredicate implements Predicate<Person> {
+    private final int incomeThreshold;
+    private final IncomeComparisonOperator incomeComparisonOperator;
+
+    /**
+     * Constructs an {@code IncomeComparisonPredicate}.
+     *
+     * @param incomeComparisonOperator The operator used to compare the person's income with the threshold.
+     * @param incomeThreshold          The threshold income to compare against.
+     */
+    public IncomeComparisonPredicate(IncomeComparisonOperator incomeComparisonOperator, int incomeThreshold) {
+        requireNonNull(incomeComparisonOperator);
+        this.incomeThreshold = incomeThreshold;
+        this.incomeComparisonOperator = incomeComparisonOperator;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        int personIncome = person.getIncome().value;
+
+        switch (incomeComparisonOperator.comparisonOperator) {
```

Comment:
If have time can consider making these greater/equal/less than behavior more type-safe (and more OOP) by implementing enums! Maybe something like this:

```
public enum IncomeComparisonOperator {
    EQUAL("="),
    GREATER_THAN(">"),
    LESS_THAN("<");
    // some assign symbol logic
}
```